File: lib/features/tasks/data/models/mission.dart
import 'package:isar/isar.dart';

part 'mission.g.dart';

@collection
class Mission {
  Id id = Isar.autoIncrement;

  late DateTime date;
  
  late String title;
  
  String? description;
  
  @enumerated
  late MissionType type;
  
  bool isCompleted = false;
  
  int xpReward = 0;
  
  int? projectId;
  
  String? reminderTime; // e.g., "09:00"
  
  String? reminderRepeatMode; // "Once", "Daily", "Weekly"
  
  List<int>? reminderWeeklyDays;
}

enum MissionType {
  main,
  side,
  routine
}

File: backend/models/Mission.js
const mongoose = require('mongoose');

const missionSchema = new mongoose.Schema({
  id: { type: Number, required: true },
  userId: { type: String, ref: 'User', required: true },
  date: { type: Date, required: true },
  title: { type: String, required: true },
  description: { type: String },
  type: { type: Number, required: true },
  isCompleted: { type: Boolean, default: false },
  xpReward: { type: Number, default: 0 },
  projectId: { type: Number }
});

module.exports = mongoose.model('Mission', missionSchema);
